addons/phone_validation/models/phone_blacklist.py

```diff
@@ -78,28 +89,27 @@ def sanitize_number(arg):
         return super()._search(domain, offset, limit, order, access_rights_uid)
 
     def add(self, number, message=None):
-        sanitized = phone_validation.phone_sanitize_numbers_w_record([number], self.env.user)[number]['sanitized']
-        return self._add([sanitized], message=message)
+        return self._add([number], message=message)
 
     def _add(self, numbers, message=None):
         """ Add or re activate a phone blacklist entry.
 
         :param numbers: list of sanitized numbers """
-        records = self.env["phone.blacklist"].with_context(active_test=False).search([('number', 'in', numbers)])
-        todo = [n for n in numbers if n not in records.mapped('number')]
-        if records:
-            if message:
-                records._track_set_log_message(message)
-            records.action_unarchive()
-        if todo:
-            new_records = self.create([{'number': n} for n in todo])
-            if message:
-                for record in new_records:
-                    record.with_context(mail_create_nosubscribe=True).message_post(
-                        body=message,
-                        subtype_xmlid='mail.mt_note',
-                    )
-            records += new_records
+
+        # Log on existing records
+        existing = self.env["phone.blacklist"].with_context(active_test=False).search([('number', 'in', numbers)])
+        if existing and message:
+            existing._track_set_log_message(message)
+
+        records = self.create([{'number': n} for n in numbers])
+
+        # Log on new records
+        if records - existing and message:
+            for record in records:
```

Comment:
We should post on new only, other should have tracking message
```suggestion
        new = records - existing
        if new and message:
            for record in new:
```

Reply:
ok, good catch, thanks! 

PS:
> We should post on new only, other should have tracking message

let me know if there is any docs/materials on tracking/posting messages

Reply:
Not sure what you mean ? But we could add a test: add two numbers, one existing (archived) and one new, check that there is a note with a tracking on the existing one and a message on the new one. That way it is correctly covered :)